src/forms/BoundSelectAndTextField.tsx

```diff
@@ -0,0 +1,37 @@
+import { CompoundField } from "src/components/internal/CompoundField";
+import { Only } from "src/Css";
+import { BoundSelectField, BoundSelectFieldProps, BoundTextField, BoundTextFieldProps } from "src/forms/index";
+import { Value } from "src/inputs/Value";
+import { TextFieldXss } from "src/interfaces";
+import { HasIdAndName, Optional } from "src/types";
+
+interface BoundSelectAndTextFieldProps<O, V extends Value, X> {
+  selectFieldProps: CompoundSelectFieldProps<O, V>;
+  textFieldProps: CompoundTextFieldProps<X>;
+  compact?: boolean;
+}
+
+export function BoundSelectAndTextField<O, V extends Value, X extends Only<TextFieldXss, X>>(
+  props: BoundSelectAndTextFieldProps<O, V, X>,
+): JSX.Element;
+export function BoundSelectAndTextField<O extends HasIdAndName<V>, V extends Value, X extends Only<TextFieldXss, X>>(
+  props: Omit<BoundSelectAndTextFieldProps<O, V, X>, "selectFieldProps"> & {
+    selectFieldProps: Optional<CompoundSelectFieldProps<O, V>, "getOptionValue" | "getOptionLabel">;
+  },
+): JSX.Element;
+export function BoundSelectAndTextField<O extends HasIdAndName<V>, V extends Value, X extends Only<TextFieldXss, X>>(
+  props: Omit<BoundSelectAndTextFieldProps<O, V, X>, "selectFieldProps"> & {
+    selectFieldProps: Optional<CompoundSelectFieldProps<O, V>, "getOptionLabel" | "getOptionValue">;
+  },
+): JSX.Element {
+  const { selectFieldProps, textFieldProps, compact = true } = props;
+  return (
+    <CompoundField>
+      <BoundSelectField {...selectFieldProps} sizeToContent compact={compact} />
```

Comment:
Hmm.. yea, custom testids are a bit of a pain here. Though, the testid for these fields is based on the field's label, so you can still customize it.
I will play around to see how we might be able to better support custom testids here.

Reply:
You _can_ do:
```
<BoundSelectAndTextField selectFieldProps={{ "data-testid": "customTestId" }} />
```
But typescript won't like that as "data-testid" isn't a BoundSelectFieldProp. But you could try instead:
```
<BoundSelectAndTextField 
  selectFieldProps={{ field: formState.type, options: types, ...{ "data-testid": "hey" } }}
/>
```
That extra spread makes Typescript happy, or really oblivious.

Reply:
This has been updated to support custom test ids.
The two fields (SelectField and TextField) will internally set a `data-testid` based on the label of the `SelectField`, which, if not given will default the `field.key` property.
In addition, you can prefix the field `data-testid`s with a custom value passed to the component, example:
```
<BoundSelectAndTextField
  selectFieldProps={{ field: formState.type, options: types }}
  textFieldProps={{ field: formState.name, clearable: true, placeholder: "Add new" }}
  data-testid="customTestId"
/>
```
This would result in the SelectField having a `data-testid="customTestId_type"` and the TextField with `data-testid="customTestId_name"`.